src/foam/nanos/auth/AgentAuthService.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.nanos.auth',
+  name: 'AgentAuthService',
+
+  documentation: `
+    The agent auth service is responsible for allowing users to act as
+    others if permitted and places the requesting user as an agent of the system within the context.
+    A junction between users are created to determine if a user is permitted to act as another user.
+    Along with the target & source Id on the junction object (UserUserJunction), a group is
+    appended and is used to determine the list of permissions for the current user agent
+    access to the system. When acting as another user, the user requesting to act is an "agent"
+    within the context. The user being logged into becomes the "user" within the context, and
```

Comment:
Should say "acted as" instead of "logged into" to keep the wording consistent and avoid confusion.